File: src/lib/src/main/java/work/lclpnet/ap2/impl/game/kit/Kit.java
package work.lclpnet.ap2.impl.game.kit;

import net.minecraft.core.RegistryAccess;
import net.minecraft.server.level.ServerPlayer;
import net.minecraft.world.item.ItemStack;

public interface Kit {

    String id();

    ItemStack createItemStack(RegistryAccess manager);

    default void init(KitOptions options) {}

    default void equip(ServerPlayer player, KitOptions options) {}

    default void unequip(ServerPlayer player, KitOptions options) {}
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/game/kit/Kit.kt
package work.lclpnet.ap2.game.kit

import net.minecraft.core.RegistryAccess
import net.minecraft.server.level.ServerPlayer
import net.minecraft.world.item.ItemStack

interface Kit {
    fun id(): String

    fun createItemStack(manager: RegistryAccess): ItemStack

    fun init(options: KitOptions) {}

    fun equip(player: ServerPlayer, options: KitOptions) {}

    fun unequip(player: ServerPlayer, options: KitOptions) {}
}
